docs/pages/architecture/introduction.mdx

```diff
@@ -0,0 +1,17 @@
+---
+title: Teleport Architecture Guides
+description: Get detailed information about how Teleport works
+---
+
+In this section, you will find detailed information about Teleport's internal
+architecture. Read these guides if you are interested in learning how Teleport
+works.
+
+- [Trusted Clusters](./trustedclusters.mdx)
```

Comment:
I'd suggest a slightly different oder:

- Authentication
- Proxy Service
- Teleport Nodes
- Session recording
- TLS routing
- Proxy peering
- Trusted clusters

Reason being:

- Trusted clusters are complicated, come with additional maintenance burden, and many of the use cases that would have required trusted clusters in the past can now be done with a more basic set up.
- This order introduces the "101" content first that is essential to understanding teleport, and moves advanced features like proxy peering after the basics.